store/criteria.js

```diff
@@ -51,13 +52,110 @@ export const mutations = {
 }
 
 export const getters = {
-  doesCriterionMatchSelection: (state, getters) => (criterion) => {
+  /**
+   * Function that checks the acceptable criteria date encoding and translates
+   * it to something that the js engine can parse / check
+   *
+   * Use Cases                                      | Required encoding (acceptable value(s))
+   * - user must be born on MM/DD/YYYY (11/14/1999) | =11-14-1999
+   * - user older than 60 years  (Y, M, D)          | >60Y
+   * - criteria before MM/DD/YYYY (01/01/2022)      | <01-01-2022
+   * - born during YYYY-YYYYY (1990-2000)           | >01-01-1990 , <01-01-2000
+   * - same use case as above (but with age range)  | >60Y, <40Y
+   * @param {currentState} state
+   * @param {storeGetters} getter
+   * @returns null / true / false [empty, pass, fail]
+   */
+  doesCriterionDateMatch: (state, getters) => (criterion) => {
```

Comment:
The store seems to be more of a data layer.  This logic feels heavy to the DateInput component.  I am not so sure it should live here.

Reply:
Sounds good will work on this tomorrow morning!

Reply:
Actually I would like to talk about this, because I originally was debating this, and thought about where in our project the input was being checked against the acceptance criteria. This only happens by calling the store, which has a little bit more than a just data storage layer I think because of the getter methods more being validation methods as well. With the large amount of logic in the `doesCriterionDateMatch` it primarily has to do with interpreting / reading the correct acceptanceValues from the markdown file. I think the logic of the dateInput should primarily contain just pulling the correct values from the user and not have a full tie to the acceptanceValues